src/diffusers/models/transformer_2d.py

```diff
@@ -228,11 +230,15 @@ def forward(
             encoder_hidden_states ( `torch.FloatTensor` of shape `(batch size, sequence len, embed dims)`, *optional*):
                 Conditional embeddings for cross attention layer. If not given, cross-attention defaults to
                 self-attention.
-            timestep ( `torch.long`, *optional*):
+            timestep ( `torch.LongTensor`, *optional*):
                 Optional timestep to be applied as an embedding in AdaLayerNorm's. Used to indicate denoising step.
             class_labels ( `torch.LongTensor` of shape `(batch size, num classes)`, *optional*):
                 Optional class labels to be applied as an embedding in AdaLayerZeroNorm. Used to indicate class labels
                 conditioning.
+            attention_mask ( `torch.Tensor` of shape (batch size, num latent pixels), *optional* ).
```

Comment:
I think I don't understand. if Unet 2d condition converted a bool tensor into a float tensor, then passed it in here..
You'd want to undo that conversion, turning it back into a bool tensor? And then what? It has to become a float tensor at some point, because the lower layers apply it as a bias. Where would that conversion now happen?

Reply:
@patrickvonplaten I'm blocked on this.

UNet2DConditionModel will send a bias (float [0, -10000]).  
other users will send a mask (int/bool/float [1, 0]).

we ultimately need to turn it into a bias (float [0, -10000]), because that's what the layers below are expecting.

so, a mask-to-bias conversion is necessary.  
but I'm unclear on what condition I can use to determine when to perform mask-to-bias conversion.

I could do it based on datatype? that kinda contradicts the decision we made in UNet2DConditionModel (don't check datatype; assume it's a mask).

I'm also unclear how to square my opinion (we need to detect masks, and convert them into biases) with your advice (we need to detect biases, and turn them into masks).

Reply:
`Transformers2DModel` is both a sub-component and a model on its own. One can do:

```py
from diffusers import Transformer2DModel

model = Transformer2DModel(...)
```

Then we using:

```py
model(..., attention_mask=attention_mask)
```

This attention_mask should not be a bias, but a boolean attention mask in which case we need to convert it manually to a bias. For this could you add this code snippet here:

```py
# prepare attention_mask
if attention_mask is not None and attention_mask.ndim == 2:
    # in case attention mask is not already converted to an attention bias let's convert it here
    attention_mask = (1 - attention_mask.to(sample.dtype)) * -10000.0
    attention_mask = attention_mask.unsqueeze(1)
```

Reply:
So to clarify, all models, such as UNet2DCondition should only accept attention masks, Transformer2DModel is an expection that can accept both mask and bias **because** it's used as a sub model of UNet2DCondition. 

So let's make sure to convert the mask into a bias if the user passes a mask to Transformer2D. We can take advantage of the fact that attention mask always has a shape [batch_size, seq_len] and bias [batch_size, 1, seq_length] => so let's just check via `attentoin_mask.ndim`

Reply:
okay, that makes more sense. I was confused by the advice "make it a boolean".

checking dims is a good idea.

I've pushed a commit that does this.